django/contrib/gis/db/backends/oracle/schema.py

```diff
@@ -119,3 +140,23 @@ def _create_spatial_index_name(self, model, field):
         return truncate_name(
             "%s_%s_id" % (strip_quotes(model._meta.db_table), field.column), 30
         )
+
+    def _queue_create_spatial_index(self, model, field):
+        """Queues SQL to drop a field's index for execution by run_geometry_sql()."""
+        self.geometry_sql.append(
+            self.sql_add_spatial_index
+            % {
+                "index": self.quote_name(self._create_spatial_index_name(model, field)),
+                "table": self.quote_name(model._meta.db_table),
+                "column": self.quote_name(field.column),
+            }
+        )
+
+    def _drop_spatial_index(self, model, field):
+        """Executes SQL to drop the spatial index on the given field."""
+        self.execute(
+            self.sql_drop_spatial_index
+            % {
+                "index": self.quote_name(self._create_spatial_index_name(model, field)),
+            }
+        )
```

Comment:
There is nothing special in dropping spatial indexes on MySQL or Oracle, so I proposed to reuse existing hooks, #17697.

Reply:
Cool, thanks. Just rebased to use that approach.